var/spack/repos/builtin/packages/qt-base/package.py

```diff
@@ -65,9 +65,13 @@ def cmake_args(self):
             if re_qt.match(dep.name):
                 qt_prefix_path.append(self.spec[dep.name].prefix)
 
-        # Now append all qt-* dependency prefixex into a prefix path
+        # Now append all qt-* dependency prefixes into a prefix path
         args.append(self.define("QT_ADDITIONAL_PACKAGES_PREFIX_PATH", ":".join(qt_prefix_path)))
 
+        # Add all dependency prefix/lib directories to RPATH
+        qt_extra_rpaths = [p.lib for p in qt_prefix_path] + [p.lib64 for p in qt_prefix_path]
+        args.append(self.define("QT_EXTRA_RPATHS", qt_extra_rpaths))
```

Comment:
@wdconinc Excellent, haampie's suggestion fixes the RPATH of the `moc` tool for me:
```suggestion
        # Make our CMAKE_INSTALL_RPATH redundant:
        # for prefix of current package ($ORIGIN/../lib type of rpaths), 
        define("QT_DISABLE_RPATH", True)
        # for prefixes of dependencies
        define("QT_NO_DISABLE_CMAKE_INSTALL_RPATH_USE_LINK_PATH", True)
```